htdocs/core/class/notify.class.php

```diff
@@ -615,7 +615,7 @@ public function send($notifcode, $object, $filename_list = array(), $mimetype_li
 
 						$ref = dol_sanitizeFileName($newref);
 						$pdf_path = $dir_output."/".$ref.".pdf";
-						if (!dol_is_file($pdf_path)) {
+						if (!dol_is_file($pdf_path)||!$arraydefaultmessage->joinfiles) {
```

Comment:
$arraydefaultmessage may still be null here (it depends on context). And if null, $arraydefaultmessage->joinfiles will generate a fatal error. So test must be enhanced to support this case...